Assets/BossRoom/Scripts/Client/UI/UnityServicesUIHandler.cs

```diff
@@ -49,25 +49,38 @@ void ServiceErrorHandler(UnityServiceErrorMessage error)
-                        $"same machine. Please use command line arg '{ProfileManager.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
-                    break;
-                }
-                case LobbyExceptionReason.LobbyFull:
-                {
-                    PopupPanel.ShowPopupPanel("Failed to join lobby", "Lobby is full and can't accept more players");
-                    // Returning out of the function because we replace default popup panel with this.
-                    return;
+                    // If the error is one of the following, the player needs to know about it, so show in a popup message. Otherwise, the log in the console is sufficient.
+                    case LobbyExceptionReason.ValidationError:
+                        PopupPanel.ShowPopupPanel("Validation Error", "Validation check failed on Lobby. Is the join code correctly formatted?");
+                        break;
+                    case LobbyExceptionReason.LobbyNotFound:
+                        PopupPanel.ShowPopupPanel("Lobby Not Found", "Requested lobby not found. The join code is incorrect or the lobby has ended.");
+                        break;
+                    case LobbyExceptionReason.LobbyConflict:
+                        // LobbyConflict can have multiple causes. Let's add other solutions here if there's other situations that arise for this.
+                        Debug.LogError($"Got service error {error.Message} with LobbyConflict. Possible conflict cause: Trying to play with two builds on the " +
+                            $"same machine. Please use command line arg '{ProfileManager.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
+                        PopupPanel.ShowPopupPanel("Failed to join Lobby", "Failed to join Lobby due to a conflict. See logs for more details.");
+                        break;
+                    case LobbyExceptionReason.NoOpenLobbies:
+                        PopupPanel.ShowPopupPanel("Failed to join Lobby", "No accessible lobbies are currently available for quick-join.");
+                        break;
+                    case LobbyExceptionReason.LobbyFull:
+                        PopupPanel.ShowPopupPanel("Failed to join Lobby", "Lobby is full and can't accept more players.");
+                        break;
+                    case LobbyExceptionReason.Unauthorized:
+                        PopupPanel.ShowPopupPanel("Lobby error", "Received HTTP error 401 Unauthorized from Lobby Service.");
+                        break;
+                    case LobbyExceptionReason.RequestTimeOut:
+                        PopupPanel.ShowPopupPanel("Lobby error", "Received HTTP error 408 Request timed out from Lobby Service.");
+                        break;
                 }
             }
-
-            PopupPanel.ShowPopupPanel("Service error: " + error.Title, errorMessage);
```

Comment:
Should we still log the unhandled ones? Or will they be handled somewhere else?

Reply:
They are already logged with the stack trace from before the async task before creating the UnityServiceErrorMessage that is published and caught here